utils.js

```diff
@@ -183,25 +183,60 @@ CTLEventUtils.readURLParams = function(eventsList, queryString, index) {
             case 'q':
                 filteredEvents = CTLEventUtils.searchEvents(
                     filteredEvents, index, splitParam[1]);
+                CTLEventUtils.populateQ(splitParam[1]);
```

Comment:
Yeah, I think that re-factoring it so the parsing of the query string and the filtering of events are separate would be a great step.